rita_client/src/operator_update/mod.rs

```diff
@@ -52,38 +61,61 @@ lazy_static! {
     static ref RITA_UPTIME: Instant = Instant::now();
 }
 
-/// Get the data to send to the operator server via websocket
-pub fn get_operator_checkin_data(
-    ops_last_seen_usage_hour: Option<u64>,
-) -> Result<OperatorCheckinMessage, RitaClientError> {
+pub fn get_operator_address() -> Option<Address> {
     let rita_client = settings::get_rita_client();
-    let id = rita_client.get_identity().unwrap();
-    let logging_enabled = rita_client.log.enabled;
     let operator_settings = rita_client.operator.clone();
-    let system_chain = rita_client.payment.system_chain;
-    let operator_address = operator_settings.operator_address;
-    let contact_info = option_convert(rita_client.exit_client.contact_info.clone());
-    let install_details = operator_settings.installation_details.clone();
-    let billing_details = operator_settings.billing_details;
-    let user_bandwidth_limit = rita_client.network.user_bandwidth_limit;
-
-    // if the user has disabled logging and has no operator configured we don't check in
-    // if the user configures an operator but has disabled logging then we assume they still
-    // want the operator to work properly and we will continue to checkin
-    if operator_address.is_none() && !logging_enabled {
-        info!("No Operator configured and logging disabled, not checking in!");
-        // return ok as this is not an error case
-        return Err(RitaClientError::MiscStringError(
-            "No Operator configured and logging disabled, not checking in!".to_string(),
-        ));
-    }
-
+    operator_settings.operator_address
+}
+pub fn get_system_chain() -> SystemChain {
+    let rita_client = settings::get_rita_client();
+    let payment = rita_client.payment;
+    payment.system_chain
+}
+pub fn get_exit_con() -> Option<ExitConnection> {
+    let rita_client = settings::get_rita_client();
+    let exit_client = rita_client.exit_client.clone();
+    // Get current exit info
+    let cur_cluster = exit_client.current_exit.clone();
+    let cur_exit = Some(CurExitInfo {
+        cluster_name: cur_cluster.clone(),
+        // Hopefully ops fills this in
+        instance_name: None,
+        instance_ip: match cur_cluster {
+            Some(a) => get_selected_exit_ip(a),
+            None => None,
+        },
+    });
+    Some(ExitConnection {
+        cur_exit,
+        client_pub_ipv6: get_client_pub_ipv6(),
+    })
+}
+pub fn get_neighbor_info() -> Vec<NeighborStatus> {
     let status = get_neighbor_status();
     let mut neighbor_info = Vec::new();
     for (_, status) in status {
         neighbor_info.push(status);
     }
-
+    neighbor_info
+}
+pub fn get_contact_info() -> Option<ContactType> {
+    let rita_client = settings::get_rita_client();
+    let exit_client = rita_client.exit_client;
+    option_convert(exit_client.contact_info)
+}
+pub fn get_install_details() -> Option<InstallationDetails> {
```

Comment:
these accessory functions would do better in the settings module, also atomic setters would be nice.

Reply:
I've put these ones that are just from settings in the settings module, but some of these getters are not easily transferable (see get_hardware_info_update) or aren't really settings (the usage/throughput getters and rita uptime) so I've left them here